go/vt/vtgate/plugin_mysql_server.go

```diff
@@ -79,6 +84,10 @@ func (vh *vtgateHandler) ComQuery(c *mysql.Conn, query string, callback func(*sq
 	// FIXME(alainjobart): Add some kind of timeout to the context.
 	ctx := context.Background()
 
+	busyLock.Lock()
+	busy[c] = true
+	busyLock.Unlock()
+
```

Comment:
Better to move this section down where the defer is called. That way, there's no risk of someone accidentally adding a return between here and there.